src/hotspot/share/oops/method.cpp

```diff
@@ -1107,14 +1107,8 @@ void Method::unlink_method() {
   _i2i_entry = Interpreter::entry_for_cds_method(methodHandle(Thread::current(), this));
   _from_interpreted_entry = _i2i_entry;
 
-  if (DynamicDumpSharedSpaces) {
-    assert(_from_compiled_entry != NULL, "sanity");
-  } else {
-    // TODO: Simplify the adapter trampoline allocation for static archiving.
-    //       Remove the use of CDSAdapterHandlerEntry.
-    CDSAdapterHandlerEntry* cds_adapter = (CDSAdapterHandlerEntry*)adapter();
-    constMethod()->set_adapter_trampoline(cds_adapter->get_adapter_trampoline());
-    _from_compiled_entry = cds_adapter->get_c2i_entry_trampoline();
+  assert(_from_compiled_entry != NULL, "sanity");
+  if (DumpSharedSpaces) {
```

Comment:
I think you can remove this if check.

Reply:
No, I can't remove the check. It doesn't work with the DynamicDumpSharedSpaces case. Note that the conditions of the assert statements are the same as the original code.

Reply:
Yes, I mess up with the two. Then it is OK, thanks for the explain.

Reply:
I sent Calvin a patch which will allow the `if (DumpSharedSpaces)` to be removed. The rest of the changes look good to me.